docs/CLIENT_SETUP.md

```diff
@@ -0,0 +1,295 @@
+# MCP Client Setup Guide
```

Comment:
I would personally document the HTTP mode and its differences with STDIO as separate from the Client documentation, for several reasons:
- The differences between HTTP and STDIO are the same across all the clients, so they do not really depend on the client.
- The client's specific documentation is mainly helpful for new users to start using the Neo4j MCP server. By adding the reading prerequisite of the HTTP mode, we're delaying the time for the new users to have their client configured with a working Neo4j MCP Server. I would personally advocate for keeping the onboarding time as low as possible.
- Adding the HTTP mode documentation will give the user a clear, contextually separated page, making it easier for new and experienced users to learn about the HTTP mode.